src/python/pants/engine/internals/graph.py

```diff
@@ -826,42 +826,42 @@ class AmbiguousCodegenImplementationsException(Exception):
     """Exception for when there are multiple codegen implementations and it is ambiguous which to
     use."""
 
-    def __init__(
-        self,
+    @classmethod
+    def create(
+        cls,
         generators: Iterable[type[GenerateSourcesRequest]],
         *,
         for_sources_types: Iterable[type[SourcesField]],
-    ) -> None:
+    ) -> AmbiguousCodegenImplementationsException:
         all_same_generator_paths = (
             len({(generator.input, generator.output) for generator in generators}) == 1
         )
         example_generator = list(generators)[0]
         input = example_generator.input.__name__
         if all_same_generator_paths:
             output = example_generator.output.__name__
-            super().__init__(
+            return cls(
                 f"Multiple of the registered code generators can generate {output} from {input}. "
                 "It is ambiguous which implementation to use.\n\nPossible implementations:\n\n"
                 f"{bullet_list(sorted(generator.__name__ for generator in generators))}"
             )
-        else:
-            possible_output_types = sorted(
-                generator.output.__name__
-                for generator in generators
-                if issubclass(generator.output, tuple(for_sources_types))
-            )
-            possible_generators_with_output = [
-                f"{generator.__name__} -> {generator.output.__name__}"
-                for generator in sorted(generators, key=lambda generator: generator.output.__name__)
-            ]
-            super().__init__(
-                f"Multiple of the registered code generators can generate one of "
-                f"{possible_output_types} from {input}. It is ambiguous which implementation to "
-                f"use. This can happen when the call site requests too many different output types "
-                f"from the same original protocol sources.\n\nPossible implementations with their "
-                f"output type:\n\n"
-                f"{bullet_list(possible_generators_with_output)}"
-            )
+        possible_output_types = sorted(
+            generator.output.__name__
+            for generator in generators
+            if issubclass(generator.output, tuple(for_sources_types))
+        )
+        possible_generators_with_output = [
+            f"{generator.__name__} -> {generator.output.__name__}"
+            for generator in sorted(generators, key=lambda generator: generator.output.__name__)
+        ]
+        return cls(
+            f"Multiple of the registered code generators can generate one of "
```

Comment:
`of the` makes the sentence sound weird. `Multiple registered code generators can ...`